### DISCUSSION TITLE
Missing Edit -> Options

### DISCUSSION THREAD

**Discussion by nalimcos** [edwardkort/WWIDesigner#66] [General]
Hello,

"Options..." seems to be missing from "Edit" menu here.

I have tried locating the problem, but couldn't find any clues.

java -jar WIDesigner-xx.jar returns no errors.

OS: Ubuntu 17.04 with MATE Desktop
JVMs tried: openjdk 8 and 9, oracle java 8
WWDesigner versions: 2.1, 2.0.1, ..... 1.0

Would you have any idea?

Thanks,
Milan

**Comment by nalimcos** [comment]
Workaround: as I expected, works properly with Win32 version of Oracle Java in WINE.

**Comment by edwardkort** [comment]
Hello Milan,

The menu layout in WIDesigner attempts to comply with the different operating-system UI standards. For Windows, preferences are found in the "Option" menu item under "Edit". For other operating systems, preferences may be found in the "Preferences" menu item under "File". I don't know what the MATE paradigm is under Linux. But just look around: it's there.

Cheers,
Edward

**Reply by friedelwolff** [reply]
Edit -> Options is missing for me as well under Linux. I tried to look really carefully all over for it, and I can't find it.

Any pointers for the relevant code where I can try to fix this or hard-wire some solution for Linux?

**Reply by edwardkort** [reply]
Please look at Discussion #84. The UI library we use in WIDesigner does not support all of the Linux windowing interfaces. Also, try invoking WIDesigner with the supplied _run_WIDesigner_java-9.cmd_ script, which sets a number of library runtime parameter.

Has any other user running Linux had success?

Cheers,
Edward Kort
Wood Wind

**Reply by friedelwolff** [reply]
I made some progress. It seems that Jidesoft relies on the environment variable `GNOME_DESKTOP_SESSION_ID` to detect whether it is running under GNOME. In my version of GNOME that variable is not set any more (I guess it was still in use with GNOME 2).

If I set that value in the environment, it launches with something closer to native widgets and my colour theme instead of the standard swing appearance, and I can get to Edit -> Preferences.

Hopefully I can now start working on the recorder files I want to create. I hope to contribute something useful.

**Reply by edwardkort** [reply]
Thank you for tracking this down. If you have a couple of minutes, would you try a couple of things:

1.  Instead of the GNOME environment variable, insert the property "set usegnome=true"
2. If you have KDE installed, instead "set usekde=true" and invoke WIDesigner under KDE.

It has been a kazillion years since I've played with UNIX systems. Hard to believe I was once a sys admin of a mid-size SUN network.

Thank you,
Edward

**Reply by friedelwolff** [reply]
I understand that I should invoke the software with `java -Dusegnome=true` or `java -Dusekde=true`. The first seems to have no effect. The second seems to have the same effect as running with the environment variable `KDE_FULL_SESSION=true`, which is what Jidesoft seems to use to detect KDE. On my system the KDE theme looks quite close to Swing, but there are a few icon differences that indicates that something changed. Importantly, this variable also causes the configuration entry on the Edit menu.

**Reply by edwardkort** [reply]
Let's see if I understand this...

1. If you are running Gnome, the only way to cleanly use WIDesigner is by setting the environment variable GNOME_DESKTOP_SESSION_ID under Linux; -Dusegnome=true in the java command line has no effect.
2. If you are running KDE, either using -Dusekde=true in the java command line or setting the environment variable KDE_FULL_SESSION=true allows you to use WIDesigner.

Do I have this right?

Thanks,
Edward

**Reply by friedelwolff** [reply]
Yes, you describe my results correctly. I should maybe also add that I was not able to use the command-line parameters from the .cmd file, since they are not supported by my installation of java (openjdk 8). Thank you for your support and for the software!

**Reply by edwardkort** [reply]
Thank you for your investigations. I will include the above in the UNIX_README in the next release.

I haven't played with the various Oracle java replacements. But if openjdk 8 is compliant with Oracle's jre 8, you shouldn't need those additional parameters. However, if/when you move to openjdk 9 or above, you probably will.

Thanks again,
Edward
